### PR TITLE
Disable Deprecated MySQL Client Reconnect

### PR DESCRIPTION
**Pull request by sureshc**
We've recently upgraded our web application server and daemon EC2 Instances from Ubuntu 18 to Ubuntu 20 and now use the MySQL 8 client. The new MySQL 8 client (which is backwards compatible with the MySQL 5.7 server that we are using) logs to Standard Output: `WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.` because we have this client feature enabled in both Pegasus and Dashboard.

Disable MySQL client reconnect because we likely don't rely on it and it generates significant noise in our logs. Specifically, anytime a cron job writes to Standard Output, our cronjob wrapper creates a HoneyBadger Error, so currently all cronjobs that connect to the database are generating a HoneyBadger Error every time that run.

- Our dashboard and pegasus web application servers connect to ProxySQL running locally on the same instance, so a connection failure is unlikely, and also the latest release of ProxySQL [doesn't support MYSQL_OPT_RECONNECT](https://proxysql.com/documentation/global-variables/mysql-variables/#mysql-default_reconnect) and the fork of an older release of ProxySQL [that we use](https://github.com/wjordan/proxysql/tree/v2.0.8) doesn't seem to use the MySQL client reconnect feature.
- [Pegasus connection adapter](https://github.com/code-dot-org/code-dot-org/blob/b43061474de78b2641495fa232938a7115e6ae58/lib/cdo/db.rb#L30-L36) and [Database connection adapter](https://github.com/code-dot-org/code-dot-org/blob/b43061474de78b2641495fa232938a7115e6ae58/dashboard/config/database.yml#L21) are configured to manage a pool of connections that automatically open new connections when existing connections in the pool are terminated/disrupted.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by Hamms — APPROVED** [review]
LGTM! Thanks for the detailed investigation notes

**Comment by sureshc** [comment]
A [unit test failed](https://drone.cdn-code.org/code-dot-org/code-dot-org/34819/1/4) due to a database connection opened by Sequel getting disconnected. I'm wondering if we need to add the `:connection_validator` [Sequel extension](http://sequel.jeremyevans.net/rdoc-plugins/files/lib/sequel/extensions/connection_validator_rb.html).

Confusingly, the unit test logs are littered with deprecation warnings about MYSQL_RECONNECT, so it appears this change does not fully disable this feature.

```
OnlyOneTest
--
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
/home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `_query': Mysql2::Error: MySQL client is not connected (Sequel::DatabaseDisconnectError)
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `block in query'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:150:in `handle_interrupt'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/mysql2-0.5.5/lib/mysql2/client.rb:150:in `query'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/adapters/mysql2.rb:142:in `block in _execute'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/database/logging.rb:38:in `log_connection_yield'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/adapters/mysql2.rb:137:in `_execute'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/adapters/utils/mysql_mysql2.rb:40:in `block in execute'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/connection_pool/threaded.rb:88:in `hold'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/database/connecting.rb:293:in `synchronize'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/adapters/utils/mysql_mysql2.rb:40:in `execute'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/dataset/actions.rb:1187:in `execute'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/adapters/mysql2.rb:281:in `execute'
from /home/circleci/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/sequel-5.67.0/lib/sequel/adapters/mysql2.rb:241:in `fetch_rows'
```

File: dashboard/config/database.yml [hunk 1]
```diff
@@ -29,10 +29,8 @@ mysql_defaults: &mysql_defaults
   # `MYSQL_OPT_CONNECT_TIMEOUT`: The connect timeout in seconds.
   connect_timeout: 2
 
-  # `MYSQL_OPT_RECONNECT`: Enable or disable automatic reconnection to the server if the connection is found to have
-  # been lost. Reconnect is off by default; this option provides a way to set reconnection behavior explicitly.
-  # See https://dev.mysql.com/doc/refman/5.7/en/c-api-auto-reconnect.html
-  reconnect: true
+  # `MYSQL_OPT_RECONNECT` is deprecated https://dev.mysql.com/doc/c-api/8.0/en/c-api-auto-reconnect.html
+  # reconnect: false
```
Comment (pablo-code-org) [inline dashboard/config/database.yml]: why not removing it completely?

File: lib/cdo/db.rb [hunk 2]
```diff
@@ -75,11 +73,15 @@ def sequel_connect(writer, reader, validation_frequency: nil, query_timeout: nil
...
 
+  # Enable read splitting with the `with_server` method.
+  # https://sequel.jeremyevans.net/rdoc-plugins/files/lib/sequel/extensions/server_block_rb.html
   db.extension :server_block
 
   # Check if validation frequency is valid and then use it to specify how often to
   # verify the db connection
   if validation_frequency && (validation_frequency == -1 || validation_frequency > 0)
+    # Handle terminated database connections.
+    # http://sequel.jeremyevans.net/rdoc-plugins/files/lib/sequel/extensions/connection_validator_rb.html
```
Comment (pablo-code-org) [inline lib/cdo/db.rb]: why is this added?

File: lib/cdo/db.rb [hunk 3]
```diff
@@ -75,11 +73,15 @@ def sequel_connect(writer, reader, validation_frequency: nil, query_timeout: nil
   end
   db = Sequel.connect writer, db_options
 
+  # Enable read splitting with the `with_server` method.
+  # https://sequel.jeremyevans.net/rdoc-plugins/files/lib/sequel/extensions/server_block_rb.html
```
Comment (pablo-code-org) [inline lib/cdo/db.rb]: why is this added?

**Review by pablo-code-org — CHANGES_REQUESTED** [review]
I am rejecting so it gets attention to the questions I just asked, not because there is anything wrong :) 

Thanks for doing this!

**Comment by Hamms** [comment]
Based on the CI errors, it seems like we will want to implement some alternative reconnection logic to replace the deprecated functionality we're disabling. I'm not sure how best to achieve that for ActiveRecord connections.

For Sequel, [the existing `validation_frequency` setting](https://github.com/code-dot-org/code-dot-org/blob/15f897a6b98051f3012b4b2b6f4c152a0ce0a899/lib/cdo/db.rb#L80-L85) is probably what we want; local testing indicates that setting that to `-1` so that we always validate connections on checkout is sufficient to get the tests passing. I'm not sure what value we want to use in production.

**Comment by sureshc** [comment]
Re-opened post git-lfs migration  in #57507

**Pull request closed without merging**